sympy/algebras/quaternion.py

```diff
@@ -247,7 +247,7 @@ def add(self, other):
         # If q2 is a number or a sympy expression instead of a quaternion
         if not isinstance(q2, Quaternion):
             if q1.real_field:
-                if q2.is_complex:
+                if q2.is_complex or q2.is_infinite:
```

Comment:
I don't think that infinite components should be accepted for quaternions. So maybe this could be left as it is.

Reply:
It seems this has been accepting infinite components till now. Should this be changed back?

Reply:
It has been accepting infinite components but that is probably not intentional. Anyway, I think that only finite components should be allowed in algebras.

Reply:
Probably quite a few places are like this. For the places where you changed `is_complex` to `is_complex or is_infinite` did you check if it makes sense for infinite quantities?

Reply:
I think that this should be left as `is_complex`